### PR TITLE
Reduce default grid header heights

### PR DESCRIPTION
**Pull request by amcclain**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [ ] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by amcclain** [comment]
We're using these values in one of our active client projects and they appear to be working well.

- I've only updated desktop for now, but would be good to review if mobile also makes sense.
- These might be a bit _too_ aggressive - I've looped in you all as reviewers as I know you've either worked directly on the grid headers or have active projects that might be doing fancy things with their headers.

File: cmp/ag-grid/AgGrid.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ AgGrid.getRowHeightForSizingMode = (mode) => (XH.isMobileApp ? AgGrid.ROW_HEIGHT
 /**
  * Header heights (in pixels)
  */
-AgGrid.HEADER_HEIGHTS = {large: 36, standard: 32, compact: 28, tiny: 22};
+AgGrid.HEADER_HEIGHTS = {large: 28, standard: 24, compact: 20, tiny: 18};
```
Comment (petradish) [inline cmp/ag-grid/AgGrid.js:101]: I like the look of large and standard, but compact and tiny feel too squished, especially if the column filter menu is enabled. I like how {compact: 22, and tiny: 20} look personally.
I also think the column filter menu icon top margin needs to be tweaked to 2px or 3px with this change: https://github.com/xh/hoist-react/blob/555a550a0b0534cb241770a87d90b07ee63c0b37/desktop/cmp/grid/impl/filter/ColumnHeaderFilter.scss#L12.

**Comment by haynesjm42** [comment]
I have no concerns with reducing the header heights, though I do agree that it is a bit aggressive - did we consider just matching the header heights to the row heights? I think it looks slightly odd for the headers to be shorter than the rows. Also might consider a different color for the header to keep them visually distinct from the rows if they are the same/similar height, I like to use `--xh-text-color-muted` for the headers in my apps

**Comment by lbwexler** [comment]
A couple of thoughts:

+ I agree this is perhaps a bit aggressive, I could see giving back 2 pixels at every size.  But I am sympathetic with the idea that it needs to be enough so it's noticeably different from the grid rows.  Also, easy for apps to adjust if needed using the constant.

+ Should/can we also reduce the size of group rows as well?  They seem to me to be more "chrome" like in most grids, and look out of place now -- see image.    Just a thought -- curious to know what others think

image

**Comment by amcclain** [comment]
Thanks - will review and reconcile the comments here as best I can as soon as I have a chance 👍

**Comment by amcclain** [comment]
- I am happy to make the reduction a bit less drastic - this was admittedly the "farthest out" proposal - I'm sure it's wise to be a bit more incremental.
- I personally like having the headers a bit smaller relative to the rows themselves - I think it helps distinguish them as "grid chrome" in favor of emphasizing the data itself. That said, one motivation I had for the really tight sizes was to make this distinction look deliberate - would want to watch out for anything that felt like an uncanny valley effect where close-but-not-the-same heights looked somehow messy or random.
- I agree with you @haynesjm42 that the muted-text coloring is nicer - we are using that in a few apps and I would be happy to have our existing `--xh-grid-header-text-color` var default changed to `--xh-text-color-muted`
- Finally, while we're considering these headers and some style changes - I'm finding myself changing `--xh-grid-header-bg` to `--xh-grid-bg` to avoid yet-another-strip-of-gray. They end up looking a bit more modern and cohesive IMO - interested if you all agree and would propose making that change too while we're at it. (Again, some flagship apps already using these settings)

**Comment by amcclain** [comment]
Updated if anyone would like to take a look for possible inclusion in Friday's release.

Lee you are right - I am eager to get those group rows slimmed down now!  (#2668)

**Comment by lbwexler** [comment]
Looking now.  Will aim to merge

**Pull request merged**